apps/extension/src/routes/page/onboarding/set-password.tsx

```diff
@@ -12,12 +12,21 @@ import { FadeTransition, PasswordInput } from '../../../components';
 import { useOnboardingSave } from '../../../hooks/onboarding';
 import { usePageNav } from '../../../utils/navigate';
 import { PagePath } from '../paths';
+import { useStore } from '../../../state';
+import { passwordSelector } from '../../../state/password';
 
 export const SetPassword = () => {
   const navigate = usePageNav();
+  const { hashedPassword } = useStore(passwordSelector);
   const finalOnboardingSave = useOnboardingSave();
   const [password, setPassword] = useState('');
   const [confirmation, setConfirmation] = useState('');
+  console.log(window.location.hash);
+
+  useEffect(() => {
+    // redirect to main route after login from other tabs or popup, needed to clean up url
+    if (hashedPassword) window.location.replace(chrome.runtime.getURL('page.html'));
+  }, [hashedPassword]);
```

Comment:
how could someone get to this page when they already have a password?

Reply:
User open tab and enter password to popup

Reply:
There we have logic, what checking is you login